### PR TITLE
Tools: add ardupilot_gazebo to install-ROS-ubuntu script - WIP

### PR DESCRIPTION
**Pull request by rmackay9**
This is a replacement for PR  https://github.com/ArduPilot/ardupilot/pull/21678 which adds gazebo to the AP-ROS install scripts.  This is all @snktshrma's work, I've raise the PR just to get over some merge issues.

The purpose of all of this of course is to make it easy to get AP and ROS running at least in SITL.

### CODE REVIEW

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 1]
```diff
@@ -209,8 +209,43 @@ heading "${green}Initializaing rosdep${reset}"
 sudo rosdep init || true
 # To find available packages, use:
 rosdep update
+
+# Installing ardupilot_gazebo
+git clone https://github.com/SwiftGust/ardupilot_gazebo
```
Comment (jmachuca77) [inline Tools/environment_install/install-ROS-ubuntu.sh:214]: this should point to https://github.com/ardupilot/ardupilot_gazebo
If that repo is missing something when we should make pull requests to that repo to bring it up to date.

File: Tools/environment_install/install-ROS-ubuntu.sh [hunk 2]
```diff
@@ -209,8 +209,43 @@ heading "${green}Initializaing rosdep${reset}"
...
+
+echo "export GAZEBO_MODEL_PATH=~/ardupilot_gazebo/models:${GAZEBO_MODEL_PATH}" >> ~/.bashrc
+echo "export GAZEBO_MODEL_PATH=~/ardupilot_gazebo/models_gazebo:${GAZEBO_MODEL_PATH}" >> ~/.bashrc
+echo "export GAZEBO_RESOURCE_PATH=~/ardupilot_gazebo/worlds:${GAZEBO_RESOURCE_PATH}" >> ~/.bashrc
+echo "export GAZEBO_PLUGIN_PATH=~/ardupilot_gazebo/build:${GAZEBO_PLUGIN_PATH}" >> ~/.bashrc
+
 # Use this to install dependencies of packages in a workspace
 # rosdep install --from-paths src --ignore-src --rosdistro=${ROS_DISTRO} -y
+
+git clone https://github.com/SwiftGust/ardupilot_gazebo
```
Comment (jmachuca77) [inline Tools/environment_install/install-ROS-ubuntu.sh:233]: this whole section is duplicated.

File: Tools/environment_install/install-ROS-ubuntu.sh (hunk 1 above)
Comment (jmachuca77) [reply Tools/environment_install/install-ROS-ubuntu.sh:214]: Actually this is incorrect... this should be cloned into the workspace and be part of the ardupilot_ws .....

**Review by jmachuca77 — CHANGES_REQUESTED** [review]
The ardupilot_ros package should form part of the ardupilot workspace. This just clones the repo into the Tools folder and compiles in there.... I'll submit a change.

File: Tools/environment_install/install-ROS-ubuntu.sh (hunk 2 above)
Comment (rmackay9) [reply Tools/environment_install/install-ROS-ubuntu.sh]: replaced with your new version.

**Comment by peterbarker** [comment]
@jmachuca77 happy with this now?

**Comment by rmackay9** [comment]
@peterbarker,

Thanks for the follow-up.  A number of issues have appeared with this script.  I've created an issue and @snktshrma is working on some fixes.  .. so for now I've changed this to a WIP.

References PRs and issues:

- https://github.com/ArduPilot/ardupilot/issues/21722
- https://github.com/ArduPilot/ardupilot/pull/21723
- https://github.com/ArduPilot/ardupilot/pull/21730

**Comment by rmackay9** [comment]
Closing in favour of https://github.com/ArduPilot/ardupilot/pull/22598

**Pull request closed without merging**